src/jsonschema/compile_evaluate.cc

```diff
@@ -169,8 +179,16 @@ class EvaluationContext {
     assert(step.relative_instance_location.size() <= 1);
     this->frame_sizes.emplace_back(step.relative_schema_location.size(),
                                    step.relative_instance_location.size());
-    this->evaluate_path_.push_back(step.relative_schema_location);
-    this->instance_location_.push_back(step.relative_instance_location);
+    Pointer relative_schema_location;
+    CONVERT_WEAK_TO_POINTER(relative_schema_location,
```

Comment:
This feels weird. I guess you do it because the instructions have `Pointer`, and you cannot push that to a `WeakPointer` yet. The solution seems to add, for `WeakPointer`, a `push_back` overload that takes non-weak pointers?